src/libcore/cell.rs

```diff
@@ -798,6 +798,15 @@ impl<'b, T: ?Sized + Unsize<U>, U: ?Sized> CoerceUnsized<RefMut<'b, U>> for RefM
 /// The `UnsafeCell<T>` type is the only legal way to obtain aliasable data that is considered
 /// mutable. In general, transmuting an `&T` type into an `&mut T` is considered undefined behavior.
 ///
+/// The compiler makes optimizations based on the knowledge that `&T` is not mutably aliased, and
```

Comment:
It should be both, holding a mutable alias to an imutably borrowed thing is unsafe (due to rearrangements even if you don't mutate then). Will fix.